clang/lib/Driver/ToolChains/Haiku.h

```diff
@@ -15,30 +15,53 @@
 
 namespace clang {
 namespace driver {
+namespace tools {
+
+/// Directly call GNU Binutils assembler and linker
+namespace haiku {
+class LLVM_LIBRARY_VISIBILITY Linker : public Tool {
```

Comment:
`final`

Reply:
I am not sure what you meant by this.

Reply:
`class LLVM_LIBRARY_VISIBILITY Linker final : public Tool`. This should improve compiler devirtualization.